pkg/vfs/writer.go

```diff
@@ -199,6 +199,9 @@ func (c *chunkWriter) commitThread() {
 			var ss = meta.Slice{Id: s.id, Size: s.length, Off: s.soff, Len: s.slen}
 			err = f.w.m.Write(meta.Background, f.inode, c.indx, s.off, ss, s.lastMod)
 			f.w.reader.Invalidate(f.inode, uint64(c.indx)*meta.ChunkSize+uint64(s.off), uint64(ss.Len))
+			if err != 0 { // Remove uploaded slices if failed to check into meta
```

Comment:
The state is not determined for some errors like "timed out".

Reply:
Whatever the state is, we no longer track it in the meta engine. So we should delete those leaked objects?

Reply:
It's possible that the `Write` actually succeeded, in which case deleting the objects will corrupt the file.

Reply:
I suppose we should reject write requests at the beginning of `fileWriter.Write` if `f.err` is already marked. Some user will always retry regardless of the error. Failing fast can protect us from those useless retries.

Reply:
This is indeed worth discussing. Currently, JuiceFS tries its best to preserve data as meticulously as possible, as the consequences of data loss are often far more severe than those of data leakage.

Reply:
Failing fast is only a way to self-protection. I suppose it has no effect on data persistence?